### PR TITLE
Sync editor upon editor change

### LINKED ISSUE #450: Creating new `.jcad` file while having another open makes the scene blank for existing one
https://github.com/user-attachments/assets/aca9f859-7058-4562-be91-153bf6617a2f

### PR DESCRIPTION
**Pull request by arjxn-py**
Should close #450

### CODE REVIEW

**Review by martinRenou — COMMENTED** [review]
Neat!

I don't quite like that this lives under `commands.ts` though. I'd suggest we put this code somewhere in the `JupyterCadPanel` class (under `packages/base/src/widget.ts`)

**Comment by trungleduc** [comment]
I agree with @martinRenou

**Comment by martinRenou** [comment]
With your recent changes you are still calling this logic in `commands.ts` which should not responsible for this IMO. I'd suggest doing this **in** the `JupyterCadPanel` class implementation, maybe in the constructor.

This may be a bit more involved as you'd need for the `tracker` to be an option for the `JupyterCadPanel` constructor, but I believe it's the way to go.

This way the JupyterCadPanel is responsible for resizing itself when the context changes. `commands.ts` is merely a place to define our commands.

**Comment by arjxn-py** [comment]
Thanks for the comprehensive explanation Martin. This sounds good, i'll be on it.

File: packages/base/src/widget.ts [hunk 1]
```diff
@@ -67,6 +69,17 @@ export class JupyterCadPanel extends SplitPanel {
     SplitPanel.setStretch(this._jupyterCadMainViewPanel, 1);
   }
 
+  private _syncEditor(tracker: WidgetTracker<IJupyterCadWidget>) {
```
Comment (martinRenou) [inline packages/base/src/widget.ts:72]: Nitpicking suggestion:

I would not define a method for this as this is called just once in the constructor. So maybe just inline that code there :)

**Review by martinRenou — COMMENTED** [review]
Neat! I just have a final suggestion

File: packages/base/src/widget.ts (hunk 1 above)
Comment (martinRenou) [reply packages/base/src/widget.ts:72]: Ah right! I agree. Sorry Arjun for the misleading comments earlier

Comment (arjxn-py) [reply packages/base/src/widget.ts:72]: No worries, It's been a good mental exercise 🤣 
I'll be on it.

Comment (arjxn-py) [reply packages/base/src/widget.ts]: Done in [`4115b79`](https://github.com/jupytercad/JupyterCAD/pull/448/commits/4115b79a5cb1a5d636ed4982df147d465364c8c4)
And I believe I myself am responsible that this PR took longer than usual as my initial iterations of solution were not close to perfect.

**Review by trungleduc — APPROVED** [review]

**Comment by trungleduc** [comment]
Thanks!

**Comment by SylvainCorlay** [comment]
Ah weird, the merge commit does not pass the integration tests.

**Comment by arjxn-py** [comment]
> Ah weird, the merge commit does not pass the integration tests.

Yes, should be related to #471

**Pull request merged**